hazelcast/src/main/java/com/hazelcast/config/CacheConfig.java

```diff
@@ -26,7 +26,7 @@
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
 import com.hazelcast.spi.impl.SerializationServiceSupport;
-import com.hazelcast.spi.merge.SplitBrainMergeTypeProvider;
+import com.hazelcast.spi.impl.merge.SplitBrainMergeTypeProvider;
```

Comment:
I recall a discussion several months ago on whether it is valid to implement private interfaces in public classes. As far as I remember, the consensus was that we should avoid that because it exposes the internal methods to public space.
By looking at the implementation of `SplitBrainMergeTypeProvider.getProvidedMergeTypes` and it's usages, I can suggest to remove that interface altogether and use a single static method which will map the config class to relevant provider. Makes sense?

Reply:
makes sense 👍

Reply:
Addressed in a separate commit. Please take a look.